src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1836,7 +1836,7 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                 else
                 {
                     var property = sourceMethod.AssociatedSymbol as SourcePropertySymbolBase;
-                    if ((object)property != null && property.IsAutoPropertyWithGetAccessor)
+                    if ((object)property != null && property.IsAutoPropertyWithGetAccessor) // PROTOTYPE: Just IsAutoProperty?
```

Comment:
>IsAutoPropertyWithGetAccessor [](http://example.com/codeflow?start=61&length=29)

We should find a way here to not bind any accessors in order to decide if we need to call ```ConstructAutoPropertyAccessorBody```. I think this question can be answered syntactically. If it is a semicolon-only accessor body, then we should call it. Otherwise, we shouldn't. Bottom line, the accessor should know that itself and there should be no need to ask anything from the property.

Reply:
I was wrong, the check is not purely syntactic. We need to consider whether the accessor is abstract/extern or declared in an interface, etc. There is a ```SourcePropertyAccessorSymbol._isAutoPropertyAccessor``` field. It looks like it should be sufficient to check its value here.